lib/thrift/binary/framed/protocol_handler.ex

```diff
@@ -11,12 +11,13 @@ defmodule Thrift.Binary.Framed.ProtocolHandler do
   }
   require Logger
 
-  @spec start_link(reference, pid, module, {module, module}) :: GenServer.on_start
+  @spec start_link(reference, pid, module, {module, module, Keyword.t}) :: GenServer.on_start
```

Comment:
Should we tighten down what `transport` is allowed to be?  Not just in the spec but also in `start_link` - `when transport in [OptionA, OptionB]`?